plugins/alerting/pkg/alerting/endpoint_receivers.go

```
@@ -0,0 +1,147 @@
+package alerting
+
+import (
+	"fmt"
+	"net/mail"
+	"net/url"
+	"strings"
+
+	cfg "github.com/prometheus/alertmanager/config"
+	"github.com/rancher/opni/pkg/alerting/shared"
+	alertingv1alpha "github.com/rancher/opni/pkg/apis/alerting/v1alpha"
+	"github.com/rancher/opni/pkg/validation"
+	"golang.org/x/exp/slices"
+)
+
+func (c *ConfigMapData) AppendReceiver(recv *cfg.Receiver) {
+	c.Receivers = append(c.Receivers, recv)
+}
+
+func (c *ConfigMapData) GetReceivers() []*cfg.Receiver {
+	return c.Receivers
+}
+
+// Assumptions:
+// - Id is unique among receivers
+// - Receiver Name corresponds with Ids one-to-one
+func (c *ConfigMapData) findReceivers(id string) (int, error) {
+	foundIdx := -1
+	for idx, r := range c.Receivers {
+		if r.Name == id {
+			foundIdx = idx
+			break
+		}
+	}
+	if foundIdx < 0 {
+		return foundIdx, fmt.Errorf("receiver with id %s not found in alertmanager backend", id)
+	}
+	return foundIdx, nil
+}
+
+func (c *ConfigMapData) UpdateReceiver(id string, recv *cfg.Receiver) error {
+	if recv == nil {
+		return fmt.Errorf("nil receiver passed to UpdateReceiver")
```

Don't check for nil here, let it panic